### PR TITLE
Prep for v1.21.0

### PR DESCRIPTION
**Pull request by odow**
## TODO

 - [x] https://github.com/jump-dev/MathOptInterface.jl/pull/2320
 - [x] #2314
 - [x] #2321
 - [x] #2322

## Basic

 - [x] `version` field of `Project.toml` has been updated
       - If a breaking change, increment the MAJOR field and reset others to 0
       - If adding new features, increment the MINOR field and reset PATCH to 0
       - If adding bug fixes or documentation changes, increment the PATCH field

## Documentation

 - [x] Add a new entry to `docs/src/changelog.md`, following existing style

## Tests

 - [x] <s>https://github.com/jump-dev/MathOptInterface.jl/actions/runs/6621005584</s>
 - [x] <s>https://github.com/jump-dev/MathOptInterface.jl/actions/runs/6622372171</s>
 - [x] <s>https://github.com/jump-dev/MathOptInterface.jl/actions/runs/6631821162</s>
 - [x] https://github.com/jump-dev/MathOptInterface.jl/actions/runs/6633588530

### CODE REVIEW

File: docs/src/changelog.md [hunk 1]
```diff
@@ -7,6 +7,31 @@ CurrentModule = MathOptInterface
...
+### Fixed
+
+ - Fixed `test_attribute_TimeLimitSec` (#2256)
+ - Fixed conversion cost in objective bridges (#2309)
+
+### Other
+
+ - Fixed flakey link checking in documentation (#2297)
+ - Minor documentation fixes (#2304) (#2313)
+ - Fixed compat of `Project.toml` (#2310)
```
Comment (odow) [inline docs/src/changelog.md:32]:
```suggestion
 - Fixed `[compat]` section of `Project.toml` (#2310)
```

**Comment by odow** [comment]
I think the PolyJuMP failure is a problem. It looks like it tried to call https://github.com/jump-dev/MathOptInterface.jl/blob/96c724f7da8d36dee1f084e16a1e64ef1631cc40/src/FileFormats/MOF/write.jl#L57

**Comment by blegat** [comment]
Yes, it seems it didn't start with an underscore before ?

File: Project.toml [hunk 2]
```diff
@@ -28,16 +28,11 @@ DataStructures = "0.18"
...
-LinearAlgebra = "1.6"
 MutableArithmetics = "1"
 NaNMath = "0.3, 1"
 OrderedCollections = "1"
 PrecompileTools = "1"
-Printf = "1.6"
-SparseArrays = "1.6"
 SpecialFunctions = "0.8, 1, 2"
-Test = "1.6"
-Unicode = "1.6"
```
Comment (odow) [inline Project.toml:40]: I've reverted these changes for now because of https://github.com/jump-dev/SCS.jl/pull/284

**Comment by odow** [comment]
Okay, I think this is now ready to go

**Review by blegat — APPROVED** [review]

**Pull request merged**